web-common/src/lib/time/ranges/formatter.ts

```diff
@@ -41,7 +39,14 @@ function formatDatePartOfTimeRange(interval: Interval, grain: V1TimeGrain) {
 
   if (showDay) format.day = "numeric";
 
-  return interval.toLocaleString(format);
+  const displayAsInclusiveEnd = grainOrder >= dayGrainOrder;
+
+  return displayAsInclusiveEnd
+    ? Interval.fromDateTimes(
```

Comment:
This needs a check to make sure start and end are not same. The test failures are because of that.